[BUG] out of tree build issue - nasm-3.01

<!-- Before making a bug report please visit https//forum.libreelec.tv where -->
<!-- a large community exists to help and support you. -->

<!-- Opening a bug report without triaging the issue in the forum first will -->
<!-- probably result in the report being closed, with an instruction to visit -->
<!-- the forum for support. -->

### Describe the bug
<!-- A clear description of what the bug is. -->
Error when building `nasm:host`. Looks to be a race condition or similar.
Regularly reproducable (50% of the time)

https://github.com/LibreELEC/actions/actions/runs/18469226571/job/52677032225

`asm` directory is not there in the `.x86_64-linux-gnu` directory. 

On logging into the GHA server and executing `scripts/build nasm:host` a second time completes the build.

### How to reproduce
Steps to reproduce the behavior:
1. Build `nasm:host` (in GHA) 


### Information
 - LibreELEC Version: 13
 - Hardware Platform: Generic/Generic-legacy

### Log file
<!-- Add debug log files (https://libreelec.wiki/how_to/provide_logfile) that we can search for errors. -->

```

Executing (host): make 
make[1]: Entering directory '/build/build.LibreELEC-Generic-legacy.x86_64-13.0-devel/build/nasm-3.01/.x86_64-linux-gnu'
/build/build.LibreELEC-Generic-legacy.x86_64-13.0-devel/toolchain/bin/host-gcc -c -Werror=attributes -march=native -O2 -Wall -pipe -I/build/build.LibreELEC-Generic-legacy.x86_64-13.0-devel/toolchain/include -Wno-format-security -std=c23 -U__STRICT_ANSI__ -ggdb3 -fwrapv -ftrivial-auto-var-init=zero -fno-common -ffunction-sections -fdata-sections -fvisibility=hidden -Wall -W -pedantic -Werror=implicit -Werror=missing-braces -Werror=return-type -Werror=trigraphs -Werror=pointer-arith -Werror=comment -Werror=vla -Werror=strict-prototypes -Werror=missing-prototypes -Werror=missing-declarations -Wno-variadic-macros -Wno-long-long -Wno-stringop-truncation -Wno-shift-negative-value -DHAVE_CONFIG_H -I/build/build.LibreELEC-Generic-legacy.x86_64-13.0-devel/build/nasm-3.01 -I. -I/build/build.LibreELEC-Generic-legacy.x86_64-13.0-devel/build/nasm-3.01/include -I./include -I/build/build.LibreELEC-Generic-legacy.x86_64-13.0-devel/build/nasm-3.01/x86 -I./x86 -I/build/build.LibreELEC-Generic-legacy.x86_64-13.0-devel/build/nasm-3.01/asm -I./asm -I/build/build.LibreELEC-Generic-legacy.x86_64-13.0-devel/build/nasm-3.01/disasm -I./disasm -I/build/build.LibreELEC-Generic-legacy.x86_64-13.0-devel/build/nasm-3.01/output -I./output -I/build/build.LibreELEC-Generic-legacy.x86_64-13.0-devel/build/nasm-3.01/zlib  -o asm/nasm.o /build/build.LibreELEC-Generic-legacy.x86_64-13.0-devel/build/nasm-3.01/asm/nasm.c
Assembler messages:
Fatal error: can't create asm/nasm.o: No such file or directory
/build/build.LibreELEC-Generic-legacy.x86_64-13.0-devel/build/nasm-3.01/asm/nasm.c:2510:1: fatal error: error writing to -: Broken pipe
 2510 | }
      | ^
compilation terminated.
make[1]: *** [Makefile:115: asm/nasm.o] Error 1

``` 
### Context
<!-- Add any context about the problem here. -->

--- a/packages/lang/nasm/patches/nasm-152-dirs-dep.patch
+++ b/packages/lang/nasm/patches/nasm-152-dirs-dep.patch
@@ -1,0 +1,22 @@
+diff --git a/Makefile.in b/Makefile.in
+index 64e0cb2d8be1..eb2adcd7e7c5 100644
+--- a/Makefile.in
++++ b/Makefile.in
+@@ -224,7 +224,7 @@ ZLIBOBJ = \
+ 
+ LIBOBJ    = $(LIBOBJ_W) $(LIBOBJ_NW) $(ZLIB)
+ ALLOBJ_W  = $(NASM) $(LIBOBJ_W)
+-ALLOBJ    = $(PROGOBJ) $(LIBOBJ)
++ALLOBJ    = $(PROGOBJ) $(LIBOBJ) $(LIBOBJ_DIS)
+ SUBDIRS  = stdlib nasmlib include config output asm disasm x86 \
+ 	   common zlib macros misc
+ XSUBDIRS = nsis win test doc editors
+@@ -262,7 +262,7 @@ ndisasm$(X): $(NDISASM) $(MANIFEST) $(DISLIB) $(NASMLIB)
+ 		$(DISLIB) $(NASMLIB) $(LIBS)
+ 
+ # Make sure we have subdirectories set up...
+-$(LIBOBJ) $(LIBOBJ_DIS): $(DIRS)
++$(ALLOBJ): $(DIRS)
+ 
+ #-- Begin Generated File Rules --#
+ 

